deepmd/pt/train/training.py

```diff
@@ -159,22 +155,43 @@ def get_data_loader(_training_data, _validation_data, _training_params):
             )
             with torch.device("cpu"):
                 training_data_buffered = BufferedIterator(iter(training_dataloader))
-            validation_dataloader = DataLoader(
-                _validation_data,
-                sampler=valid_sampler,
-                batch_size=None,
-                num_workers=min(NUM_WORKERS, 1),
-                drop_last=False,
-                pin_memory=True,
-            )
-
-            with torch.device("cpu"):
-                validation_data_buffered = BufferedIterator(iter(validation_dataloader))
-            if _training_params.get("validation_data", None) is not None:
-                valid_numb_batch = _training_params["validation_data"].get(
-                    "numb_btch", 1
+            if _validation_data is not None:
```

Comment:
the same operations applied to both train and validation. consider writting a function for them rather than copying code.